components/omega/src/ocn/Eos.cpp

```diff
@@ -0,0 +1,254 @@
+//       SpecVolFldName.append(Name);
+//       SpecVolDisplacedFldName.append(Name);
+//    }
+//
+//    // Create fields for state variables
+//    int NDims = 2;
+//    std::vector<std::string> DimNames(NDims);
+//    DimNames[0] = "NCells";
+//    DimNames[1] = "NVertLevels";
+//
+//    auto SpecVolField =
+//        Field::create(SpecVolFldName,               // Field name
+//                      "Specific Volume of layer on cell center", /// long Name
+//                      "m3 kg-1",                                 // units
+//                      "sea_water_specific_volume",         // CF standard-ish
+//                      Name 0.0,                                 // min valid
+//                      value 9.99E+30,                            // max valid
+//                      value -9.99E+30, // scalar used for undefined entries
+//                      NDims,     // number of dimensions
+//                      DimNames   // dimension names
+//        );
+//    auto SpecVolDisplacedField =
+//        Field::create(SpecVolDisplacedFldName,               // Field name
+//                      "Specific Volume displaced adiabatically to 1 layer
+//                      below", /// long Name "m3 kg-1", // units
+//                      "sea_water_specific_volume_displaced", // CF
+//                      standard-ish Name 0.0, // min valid value 9.99E+30, //
+//                      max valid value -9.99E+30, // scalar used for undefined
+//                      entries NDims,     // number of dimensions DimNames   //
+//                      dimension names
+//        );
+//
+//    // Create a field group for the eos-specific state fields
+//    EosGroupName = "Eos";
+//    if (Name != "Default") {
+//       EosGroupName.append(Name);
+//    }
+//    auto EosGroup = FieldGroup::create(EosGroupName);
+//
+//    // Add restart group if needed
+//    if (!FieldGroup::exists("Restart"))
+//       auto RestartGroup = FieldGroup::create("Restart");
+//
+//    Err = EosGroup->addField(SpecVolDisplacedFldName);
+//    if (Err != 0)
+//       LOG_ERROR("Error adding {} to field group {}", SpecVolDisplacedFldName,
+//                 EosGroupName);
+//    Err = EosGroup->addField(SpecVolFldName);
+//    if (Err != 0)
+//       LOG_ERROR("Error adding {} to field group {}", SpecVolFldName,
+//                 EosGroupName);
+//
+//    Err = FieldGroup::addFieldToGroup(SpecVolDisplacedFldName, "Restart");
+//    if (Err != 0)
+//       LOG_ERROR("Error adding {} to Restart field group",
+//                 SpecVolDisplacedFldName);
+//    Err = FieldGroup::addFieldToGroup(SpecVolFldName, "Restart");
+//    if (Err != 0)
+//       LOG_ERROR("Error adding {} to Restart field group",
+//                 SpecVolFldName);
+//
+//    // Associate Field with data
+//    I4 TimeIndex;
+//    Err = getTimeIndex(TimeIndex, 0);
+//
+//    Err =
+//        SpecVolDisplacedField->attachData<Array2DReal>(SpecVolDisplaced[TimeIndex]);
```

Comment:
I think one issue may be that `SpecVol` doesn't have a time index like variables in `State`